packages/core/src/App/Containers/AccountSwitcher/__tests__/account-switcher.spec.jsx

```diff
@@ -0,0 +1,178 @@
+        },
+        {
+            loginid: 'VRTC4444904',
+            is_disabled: 0,
+            is_virtual: 1,
+            icon: 'real',
+            title: 'Real',
+        },
+    ],
+    client_residence: 'id',
+    country_standpoint: { is_united_kingdom: true },
+    is_dark_mode_on: false,
+    is_eu: false,
+    is_landing_company_loaded: true,
+    is_low_risk: true,
+    is_high_risk: false,
+    is_logged_in: true,
+    is_virtual: true,
+    has_fiat: true,
+    landing_company_shortcode: 'svg',
+    mt5_login_list: [
+        {
+            account_type: 'demo',
+            sub_account_type: 'financial_stp',
+        },
+    ],
+    obj_total_balance: { amount_real: 5000, amount_mt5: 300, amount_dxtrade: 100, currency: 'USD' },
+    switchAccount: jest.fn(),
+    resetVirtualBalance: jest.fn(),
+    has_active_real_account: true,
+    logoutClient: jest.fn(),
+    upgradeable_landing_companies: ['maltainvest'],
+    is_positions_drawer_on: false,
+    openRealAccountSignup: jest.fn(),
+    toggleAccountsDialog: jest.fn(),
+    togglePositionsDrawer: jest.fn(),
+    toggleSetCurrencyModal: jest.fn(),
+    should_show_real_accounts_list: true,
+    real_account_creation_unlock_date: 2022,
+    setShouldShowCooldownModal: jest.fn(),
+    show_eu_related_content: false,
+    content_flag: 'LOW_RISK_CR_NON_EU',
+    has_any_real_account: true,
+    virtual_account_loginid: 'VRTC123123',
+    setTogglePlatformType: jest.fn(),
+};
+
+describe('AccountSwitcher', () => {
+    const renderWithRouter = children => {
+        render(<BrowserRouter>{children}</BrowserRouter>);
+    };
+
+    it('should render account switcher', () => {
+        renderWithRouter(<AccountSwitcher {...mock_connect_props} />);
+        expect(screen.getByTestId('acc-switcher')).toBeInTheDocument();
+        expect(screen.getByText('Real')).toBeInTheDocument();
+    });
+
+    it('should render a loader when landing companies are not loaded', () => {
+        renderWithRouter(<AccountSwitcher {...mock_connect_props} is_landing_company_loaded={false} />);
+        expect(screen.getByText('mockedLoading')).toBeInTheDocument();
+        expect(screen.queryByText('Real')).not.toBeInTheDocument();
+    });
+
+    it('should list the real accounts in real tab when landing companies are loaded', () => {
+        renderWithRouter(<AccountSwitcher {...mock_connect_props} />);
+        expect(screen.queryByText('mockedLoading')).not.toBeInTheDocument();
+        expect(screen.getByText('Real')).toHaveClass('dc-tabs__active');
+        expect(screen.getByText('CR5241840')).toBeInTheDocument();
+    });
+
+    it('should list the demo accounts in demo tab when landing companies are loaded', () => {
+        renderWithRouter(<AccountSwitcher {...mock_connect_props} should_show_real_accounts_list={false} is_virtual />);
+        expect(screen.queryByText('mockedLoading')).not.toBeInTheDocument();
+        expect(screen.getAllByText('Demo')[0]).toHaveClass('dc-tabs__active');
```

Comment:
Is there any reason for using getAllByText() here instead of getByText() as in the case of real account?

Reply:
@shaheer-deriv In demo tab, Tab-header and Account header is "Demo". So we have multiple "Demo".
In real tab, Tab-header is Real and Account header is its corresponding currency. Thus we had only one "Real" text.